### PR TITLE
fix: revert fixes for access_denied error handling for OAuth destinations (#3853)

### PR DESCRIPTION
**Pull request by sanpj2292**
# Description

This reverts commit 0d30d3b37fa1bef9a7cd8d11fd2deb9f481c3a51.
Reverting due to minor issues found while testing.

## Linear Ticket

https://linear.app/rudderstack/issue/INT-367/testing-review-and-release

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by fracasula — APPROVED** [review]

**Review by lvrach — APPROVED** [review]

**Comment by sanpj2292** [comment]
> testing should be part of the pull request and not an afterthought

It was an unexpected one not an afterthought but will keep it in-mind

**Comment by fracasula** [comment]
> > testing should be part of the pull request and not an afterthought
> 
> It was an unexpected one not an afterthought but will keep it in-mind

I think what @atzoum meant was to test it on dev before merging. As an alternative there is the "Do not merge" labels if someone wants to ask for review but has not tested it fully.

**Pull request merged**
